addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -76,7 +76,8 @@ def done(self):
                 if (picking_type.use_create_lots or picking_type.use_existing_lots):
                     for ml in picking.move_line_ids:
                         if ml.product_id.tracking != 'none':
-                            raise UserError(_('Some products require lots/serial numbers.'))
+                            if not ml.lot_name and not ml.lot_id:
+                                raise UserError(_('Some products require lots/serial numbers.'))
```

Comment:
Another option is remove code from line 74 `# If no lots when needed, raise error` and let the stock move line code show the error when necessary:
https://github.com/odoo/odoo/blob/5969d22a93a379b61ba604f10b379f97225149be/addons/stock/models/stock_move_line.py#L454-L455